packages/favorites/src/lib/Favorites.tsx

```diff
@@ -21,53 +21,62 @@ const FavoritesContext = createContext<FavoritesContextType | null>(null);
 export const FavoritesProvider = ({
   children,
   onFavoritesChange,
+  onMaxFavoritesReached,
   settingsFavorites,
   settingsStatus,
+  applicationId = NAV_APP_ID,
+  maxFavorites = MAX_FAVORITES,
 }: {
   children: React.ReactNode;
   onFavoritesChange?: (favorites: Favorite[]) => void;
+  onMaxFavoritesReached?: (favorites: Favorite[]) => void;
   settingsFavorites?: Favorite[];
   settingsStatus?: StatusUnion;
+  applicationId?: string;
+  maxFavorites?: number;
 }): JSX.Element => {
   const [lastClickedFavoriteId, setLastClickedFavoriteId] = useState<string>('');
 
   const queryClient = useQueryClient();
-  const { data: favoritesData, status: favoritesDataStatus } = useFavoritesQuery(!settingsStatus);
+  const { data: favoritesData, status: favoritesDataStatus } = useFavoritesQuery(!settingsStatus, applicationId);
   const favorites = settingsStatus ? settingsFavorites : favoritesData;
   const queryStatus = settingsStatus ? settingsStatus : favoritesDataStatus;
 
   const { submitFavorites, status: mutationStatus } = useSubmitFavorites({
     onMutationStart(targetFavoriteId) {
       setLastClickedFavoriteId(targetFavoriteId);
     },
-  });
+  }, applicationId);
 
   useEffect(() => {
-    const unsubscribeFavoritesChanged = avMessages.subscribe(
-      AV_INTERNAL_GLOBALS.FAVORITES_CHANGED,
-      (data) => {
-        if (data?.favorites) {
-          queryClient.setQueryData(['favorites'], data?.favorites);
-        }
-      },
-      { ignoreSameWindow: false }
-    );
-
-    const unsubscribeFavoritesUpdate = avMessages.subscribe(
-      AV_INTERNAL_GLOBALS.FAVORITES_UPDATE,
-      (data) => {
-        if (data?.favorites) {
-          queryClient.setQueryData(['favorites'], data?.favorites);
-        }
-      },
-      { ignoreSameWindow: false }
-    );
-
-    return () => {
-      unsubscribeFavoritesChanged();
-      unsubscribeFavoritesUpdate();
-    };
-  }, [queryClient]);
+    if (applicationId === NAV_APP_ID) {
+      const unsubscribeFavoritesChanged = avMessages.subscribe(
+        AV_INTERNAL_GLOBALS.FAVORITES_CHANGED,
+        (data) => {
+          if (data?.favorites) {
+            queryClient.setQueryData(['favorites'], data?.favorites);
+          }
+        },
+        { ignoreSameWindow: false }
+      );
+
+      const unsubscribeFavoritesUpdate = avMessages.subscribe(
+        AV_INTERNAL_GLOBALS.FAVORITES_UPDATE,
+        (data) => {
+          if (data?.favorites) {
+            queryClient.setQueryData(['favorites'], data?.favorites);
+          }
+        },
+        { ignoreSameWindow: false }
+      );
+
+      return () => {
+        unsubscribeFavoritesChanged();
+        unsubscribeFavoritesUpdate();
+      };
+    }
+    return () => null;
```

Comment:
I feel like this return shouldn't be needed since it isn't really doing anything

Reply:
resolved via https://github.com/Availity/element/pull/694/commits/d28c27200890c66906df1c67b73943383d572524